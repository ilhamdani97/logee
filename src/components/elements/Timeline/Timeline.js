import React from 'react';
import styles from './styles.scoped.css';
import moment from 'moment';
import line from '../../../assets/draw-timeline.svg';
import dot from '../../../assets/ic-timeline-dot.svg';
import { STATUS_DELIVERY_MESSAGE } from '@utils/constants';
import PropTypes from 'prop-types';
import compareProps from '@utils/compareProps';
import { formatFullDate } from '@utils/format';

function Timeline({ code, date, destinationId, isFirst }) {
  return (
    <div className={styles.content}>
      <img src={isFirst ? dot : line} />
      <div>
        <p>{destinationId ? `${STATUS_DELIVERY_MESSAGE[code].label} ${destinationId}` :
          STATUS_DELIVERY_MESSAGE[code].label}</p>
        <p>{date && `${formatFullDate(date, 2)}, ${moment(date).format('HH.mm')}`}</p>
      </div>
    </div>
  );
}

export default React.memo(Timeline, compareProps);

Timeline.defaultProps = {
  code: '',
  date: '',
  destinationId: '',
  isFirst: true,
  name: ''
};

Timeline.propTypes = {
  code: PropTypes.string,
  date: PropTypes.string,
  destinationId: PropTypes.string,
  isFirst: PropTypes.bool,
  name: PropTypes.string
};
