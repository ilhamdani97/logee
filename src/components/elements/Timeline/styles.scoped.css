.content {
  display: flex;
  align-items: flex-start;
}


.content > img {
  margin-bottom: 0.25rem;
  width: 0.75rem;
}

.content > div {
  display: flex;
  flex-direction: column;
}

.content > div > p {
  margin-left: 1.3125rem;
  margin-bottom: 0.5rem;
}

.content > div > p:last-child {
  margin-bottom: 0;
}

.content > div > p:first-child {
  font-weight: bold;
  font-size: 1.125rem;
}

.content > div > p:last-child {
  font-size: 0.75rem;
  font-weight: normal;
  color: #A0A4A8;
}