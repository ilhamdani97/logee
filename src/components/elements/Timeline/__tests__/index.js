import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Timeline from '../Timeline';

const { defaultProps } = Timeline.type;

jest.mock('../../../../utils/constants', () => ({
  STATUS_DELIVERY_MESSAGE: {
    1: { label: 'Menunggu Konfirmasi', icon: 'ic-hourglass' }
  },
}));

jest.mock('../../../../utils/storage', () => ({
  getToken: () => ''
}));

describe('src/components/elements/Timeline', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Timeline {...defaultProps} code="1" />);
    expect(tree).toMatchSnapshot();
  });

  test('render with props', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(
      <Timeline
        {...defaultProps}
        code="1"
        date={'2020-11-24'}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
