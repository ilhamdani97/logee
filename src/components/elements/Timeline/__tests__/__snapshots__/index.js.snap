// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Timeline render 1`] = `
<div>
  <img
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div>
    <p>
      Menunggu Konfirmasi
    </p>
    <p>
      
    </p>
  </div>
</div>
`;

exports[`src/components/elements/Timeline render with props 1`] = `
<div>
  <img
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div>
    <p>
      Menunggu Konfirmasi
    </p>
    <p>
      DD MMM YY, HH.mm
    </p>
  </div>
</div>
`;
